genie-core/src/test/java/com/netflix/genie/core/jobs/workflow/impl/InitialSetupTaskUnitTest.java

```diff
@@ -37,23 +48,165 @@
 @Category(UnitTest.class)
 public class InitialSetupTaskUnitTest {
 
-    private Registry registry;
     private InitialSetupTask initialSetupTask;
+    private Path temporaryTestDirectory;
 
     /**
      * Setup to run before each test.
+     * @throws IOException if a temporary folder cannot be created
      */
     @Before
-    public void setUp() {
-        this.registry = Mockito.mock(Registry.class);
+    public void setUp() throws IOException {
+        final Registry registry = Mockito.mock(Registry.class);
         this.initialSetupTask = new InitialSetupTask(registry);
+        this.temporaryTestDirectory = Files.createTempDirectory(this.getClass().getSimpleName() + "-");
```

Comment:
Changed to use @Rule. Not sure what I was doing wrong before, but copying from a different test worked without problems.